examples/models/ResNet50.cpp

```diff
@@ -0,0 +1,1293 @@
+    //     do_reorder(eng, net, net_args, weights_memory, weights_memory_nchw);
+    if (with_bias) write_to_dnnl_memory(bias_data, bias_memory);
+
+    auto src_md = src_memory.get_desc();
+    memory::desc bias_md = {};
+    if (with_bias) bias_md = bias_memory.get_desc();
+    auto weights_md = weights_memory.get_desc();
+    //     auto weights_md_nchw = memory::desc(weights_md.dims(), dt::f32,
+    //     tag::nchw);
+    auto dst_md = dst_memory.get_desc();
+
+    auto conv1_desc = convolution_forward::desc(prop_kind::forward_inference,
+            algorithm::convolution_direct, src_md, weights_md, bias_md, dst_md,
+            strides, padding, padding);
+    auto prim_desc = convolution_forward::primitive_desc(conv1_desc, eng);
+
+    net.push_back(convolution_forward(prim_desc));
+    net_args.push_back({{DNNL_ARG_SRC, src_memory},
+            {DNNL_ARG_WEIGHTS, weights_memory}, {DNNL_ARG_DST, dst_memory}});
+}
+
+void do_relu(engine &eng, std::vector<primitive> &net,
+        std::vector<std::unordered_map<int, memory>> &net_args,
+        memory &src_dst_mem) {
+    const float slope = 0.0f;
+    auto relu1_desc = eltwise_forward::desc(prop_kind::forward_inference,
+            algorithm::eltwise_relu, src_dst_mem.get_desc(), slope);
+    auto relu1_prim_desc = eltwise_forward::primitive_desc(relu1_desc, eng);
+    net.push_back(eltwise_forward(relu1_prim_desc));
+    net_args.push_back(
+            {{DNNL_ARG_SRC, src_dst_mem}, {DNNL_ARG_DST, src_dst_mem}});
+}
+
+memory::desc do_input_padding(memory &input, std::vector<float> &new_data,
+        memory::dims &padding_l, memory::dims &padding_r) {
+    auto md = input.get_desc();
+    auto new_dims = md.dims();
+    new_dims[2] += (padding_l[0] + padding_r[0]);
+    new_dims[3] += (padding_l[1] + padding_r[1]);
+    std::vector<float> old_data(product(md.dims()));
+    read_from_dnnl_memory(old_data.data(), input);
+    new_data.resize(product(new_dims));
+    const size_t old_h = md.dims()[2];
+    const size_t old_w = md.dims()[3];
+    const size_t new_h = new_dims[2];
+    const size_t new_w = new_dims[3];
+    size_t N = md.dims()[0], C = md.dims()[1], H = md.dims()[2],
+           W = md.dims()[3];
+    size_t new_N = new_dims[0], new_C = new_dims[1], new_H = new_dims[2],
+           new_W = new_dims[3];
+    for (size_t n = 0; n < N; n++) {
+        for (size_t h = 0; h < H; h++) {
+            for (size_t w = 0; w < W; w++) {
+                for (size_t c = 0; c < C; c++) {
+
+                    size_t old_ofs = n * H * W * C + h * W * C + w * C + c;
+                    size_t new_ofs = n * new_C * new_H * new_W
+                            + (h + padding_l[0]) * new_W * new_C
+                            + (w + padding_l[1]) * new_C + c;
+
+                    new_data[new_ofs] = old_data[old_ofs];
+                }
+            }
+        }
+    }
+
+    // Check correctness
+
+    //     for (size_t n = 0; n < new_N; n++) {
```

Comment:
How about checking for a macro `ONEDNN_CHECK_CORRECTNESS` or similar instead of having a commented out block of code?
It is strange to see that only this function checks for correctness, maybe this should simply be removed?

Reply:
You are right, we have a function which does the correctness check. Removed this commented code.